src/settings/__tests__/settingsReducer-test.js

```diff
@@ -121,4 +125,52 @@ describe('settingsReducer', () => {
       expect(actualState).toEqual(expectedState);
     });
   });
+
+  describe('EVENT', () => {
+    describe('type `user_settings`, op `update`', () => {
+      const eventCommon = { id: 0, type: EventTypes.user_settings, op: 'update' };
+
+      const mkCheck = <S: $Keys<SettingsState>, E: $Keys<UserSettings>>(
+        statePropertyName: S,
+        eventPropertyName: E,
+      ): ((
+        $ElementType<SettingsState, S>,
+        $ElementType<UserSettings, E>,
+        $ElementType<SettingsState, S>,
+      ) => void) => (initialStateValue, eventValue, expectedStateValue) => {
+        const initialState = { ...baseState };
+        /* $FlowFixMe[incompatible-type]: Trust that the caller passed the
+           right kind of value for its chosen key. */
+        initialState[statePropertyName] = initialStateValue;
+
+        const expectedState = { ...initialState };
+        /* $FlowFixMe[incompatible-type]: Trust that the caller passed the
+           right kind of value for its chosen key. */
+        expectedState[statePropertyName] = expectedStateValue;
+
+        expect(
+          settingsReducer(initialState, {
+            type: EVENT,
+            event: { ...eventCommon, property: eventPropertyName, value: eventValue },
+          }),
+        ).toEqual(expectedState);
+      };
+
+      describe('offlineNotification / enable_offline_push_notifications', () => {
+        const check = mkCheck('offlineNotification', 'enable_offline_push_notifications');
+        check(true, true, true);
+        check(true, false, false);
+        check(false, true, true);
+        check(false, false, false);
```

Comment:
The expected state value should always equal the event value, right? After all the [doc](https://zulip.com/api/get-events#user_settings-update) says:
> value: boolean | integer | string
> New value of the changed setting.

So I think best to encode that fact right into `mkCheck`. It makes it easier to see that the tests aren't doing anything funny; and it should make the code simpler, too.